pandas/parser.pyx

```diff
@@ -437,7 +440,11 @@ cdef class TextReader:
         # suboptimal
         if usecols is not None:
             self.has_usecols = 1
-            self.usecols = set(usecols)
+            if callable(usecols):
+                self.callable_usecols = 1
```

Comment:
I would not use this additional variable and instead just do a ``callable(self.usecalls)`` I think its more obvious